src/services/identity/LaunchClient.ts

```diff
@@ -1,26 +1,41 @@
 import {
   AmplifyClient,
   CreateDomainAssociationCommand,
-  CreateDomainAssociationCommandInput,
+  CreateDomainAssociationCommandInput as AmplifySDKCreateDomainAssociationCommandInput,
   CreateDomainAssociationCommandOutput,
   GetDomainAssociationCommand,
-  GetDomainAssociationCommandInput,
+  GetDomainAssociationCommandInput as AmplifySDKGetDomainAssociationCommandInput,
   GetDomainAssociationCommandOutput,
   SubDomainSetting,
 } from "@aws-sdk/client-amplify"
-import { options } from "joi"
+import { SubDomain } from "aws-sdk/clients/amplify"
 
-import logger from "@root/logger/logger"
+import { config } from "@config/config"
 
+// stricter typing to interact with Amplify SDK
+type CreateDomainAssociationCommandInput = {
```

Comment:
fyi, if the intention here is to also make all the keys required, then you'd need to add `Required` around the curly braces.

otherwise, this would still allow `undefined` for optional keys. (can be seen by mousing over)

Reply:
nope, got some keys that we dont need to configure! I was thinking we can be stricter with our typing if we need to modify those values in the future!